### PR TITLE
core: added documentation and assert to cib

### PR DESCRIPTION
**Pull request by OlegHahm**
Adopted #4086 from @IldarValiev

### CODE REVIEW

**Comment by Kijewski** [comment]
"Should be 0 or a power of 2" might be easier to read, because `^` could be mistaken for xor. Though "2 xor n" makes no sense.
ACK in either case.

**Comment by OlegHahm** [comment]
Addressed @Kijewski's comment and squashed immediately.

**Comment by Kijewski** [comment]
RIOT-CI fails because of #4987.

File: core/include/cib.h [hunk 1]
```diff
@@ -45,9 +47,13 @@ typedef struct {
  * @param[out] cib      Buffer to initialize.
  *                      Must not be NULL.
  * @param[in]  size     Size of the buffer, must not exceed MAXINT/2.
+ *                      Should be equal to 0 or power of 2.
  */
 static inline void cib_init(cib_t *__restrict cib, unsigned int size)
 {
+    /* check if size is a power of 2 by comparing it to its complement */
+    assert(!(size & (size - 1)));
+
```
Comment (authmillenon) [inline core/include/cib.h]: Travis fails because of trailing whitespace here.

**Comment by OlegHahm** [comment]
@kaspar030, care to ACK?

**Comment by kaspar030** [comment]
ACK!

**Comment by authmillenon** [comment]
And go

**Pull request merged**